### PR TITLE
added moveit yaml files to cob_hardware_config

### PR DESCRIPTION
**Pull request by NadiaHG**

### CODE REVIEW

**Comment by NadiaHG** [comment]
requires https://github.com/ipa320/cob_manipulation/pull/86/files

File: cob_bringup/robots/cob4-1.xml [hunk 1]
```diff
@@ -45,8 +46,8 @@
         <include file="$(find cob_bringup)/drivers/sick_flexisoft.launch">
             <arg name="robot" value="$(arg robot)"/>
         </include>
-        <!--include file="$(find cob_bringup)/tools/collision_monitor.launch">
-        </include-->
+        <include file="$(find cob_bringup)/tools/collision_monitor.launch">
```
Comment (floweisshardt) [inline cob_bringup/robots/cob4-1.xml]: I'd like to keep this commented out as long as it is well tested on HW. nevertheless we should merge the rest of this PR.

File: cob_bringup/robots/cob4-2.xml [hunk 2]
```diff
@@ -45,8 +46,8 @@
         <include file="$(find cob_bringup)/drivers/sick_flexisoft.launch">
             <arg name="robot" value="$(arg robot)"/>
         </include>
-        <!--include file="$(find cob_bringup)/tools/collision_monitor.launch">
-        </include-->
+        <include file="$(find cob_bringup)/tools/collision_monitor.launch">
```
Comment (floweisshardt) [inline cob_bringup/robots/cob4-2.xml]: I'd like to keep this commented out as long as it is well tested on HW. nevertheless we should merge the rest of this PR.

File: cob_bringup/robots/cob4-3.xml [hunk 3]
```diff
@@ -44,9 +44,6 @@
 			<arg name="robot" value="$(arg robot)" />
 		</include>
 		<include file="$(find cob_script_server)/launch/script_server.launch" />
-		<!--include file="$(find cob_bringup)/tools/base_collision_observer.launch" >
```
Comment (floweisshardt) [inline cob_bringup/robots/cob4-3.xml:47]: @ipa-nhg: why did you remove the base_collision_observer? IMHO this does something different than the collision_monitor: providing the safe twist command topic for the base

File: cob_bringup/robots/cob4-4.xml [hunk 4]
```diff
@@ -40,9 +40,6 @@
 			<arg name="robot" value="$(arg robot)" />
 		</include>
 		<include file="$(find cob_script_server)/launch/script_server.launch" />
-		<!--include file="$(find cob_bringup)/tools/base_collision_observer.launch" >
```
Comment (floweisshardt) [inline cob_bringup/robots/cob4-4.xml:43]: see comment cob4-3

File: cob_bringup/robots/cob4-6.xml [hunk 5]
```diff
@@ -40,9 +40,6 @@
 			<arg name="robot" value="$(arg robot)" />
 		</include>
 		<include file="$(find cob_script_server)/launch/script_server.launch" />
-		<!--include file="$(find cob_bringup)/tools/base_collision_observer.launch" >
```
Comment (floweisshardt) [inline cob_bringup/robots/cob4-6.xml:43]: see comment cob4-3

File: cob_hardware_config/cob3-2/config/joint_limits.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# joint_limits.yaml allows the dynamics properties specified in the URDF to be overwritten or augmented as needed
```
Comment (floweisshardt) [inline cob_hardware_config/cob3-2/config/joint_limits.yaml]: I assume this has been copied from cob_moveit_config. So we don't need to review in detail. @ipa-nhg: right?

File: cob_hardware_config/common/upload_robot.launch [hunk 7]
```diff
@@ -1,13 +1,34 @@
 <?xml version="1.0"?>
 <launch>
 
-    <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
-    <arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
+	<arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
+	<arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
+	<arg name="robot_description" default="robot_description"/>
+	<arg name="config_path" default="$(find cob_hardware_config)/$(arg robot)/config"/>
```
Comment (floweisshardt) [inline cob_hardware_config/common/upload_robot.launch]: no need for this launch arg: we can simply ise the pkg_hardware_config as we do for all other entries in cob_bringup

File: cob_hardware_config/common/upload_robot.launch [hunk 8]
```diff
@@ -1,13 +1,34 @@
 <?xml version="1.0"?>
 <launch>
 
-    <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
-    <arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
+	<arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
+	<arg name="pkg_hardware_config" default="$(find cob_hardware_config)"/>
+	<arg name="robot_description" default="robot_description"/>
```
Comment (floweisshardt) [inline cob_hardware_config/common/upload_robot.launch]: what's the use case for naming this different than robot_description? If there is no use case, we should remove that argument

File: cob_hardware_config/common/upload_robot.launch (hunk 7 above)
Comment (NadiaHG) [inline cob_hardware_config/common/upload_robot.launch]: you are right , I though that planning_context needed it https://github.com/ipa-nhg/cob_manipulation/blob/Fix/MoveitDepends/cob_moveit_config/launch/planning_context.xml#L5 but it use the same path , only the package name is different

File: cob_bringup/robots/cob4-1.xml (hunk 1 above)
Comment (NadiaHG) [inline cob_bringup/robots/cob4-1.xml]: OK! I will do it on a separate PR

File: cob_bringup/robots/cob4-2.xml (hunk 2 above)
Comment (NadiaHG) [inline cob_bringup/robots/cob4-2.xml]: OK! I will do it on a separate PR

File: cob_hardware_config/cob3-2/config/joint_limits.yaml (hunk 6 above)
Comment (NadiaHG) [inline cob_hardware_config/cob3-2/config/joint_limits.yaml]: yep! maybe the files on cob_moveit_config can be deleted, I am not sure because cob_moveit_config is a package generated automatically by moveit_setup_assistant maybe in case that we need to modify something using the assistant and we want to load our current model  is required that the files are there(?)

File: cob_bringup/robots/cob4-3.xml (hunk 3 above)
Comment (NadiaHG) [inline cob_bringup/robots/cob4-3.xml:47]: it was a mistake , I wanted to remove only the collision_monitor

**Comment by fmessmer** [comment]
Well, I'd have a few comments but typing on the phone on the small screen is to much trouble...just make sure both collision_monitoring AND moveit, i.e. planning e.g. via rviz still work afterwards!

Also cob_moveit_config pkg is not fully auto-generated...it's structure has been cob-timized!
@ipa-mdl

**Comment by fmessmer** [comment]
maybe yoj could at least keep the part uploading tbe moveit info in its own launch file similar to planning_context.xml so it is clear where it comes from...

**Comment by fmessmer** [comment]
How about keeping the moveit_config files grouped in a subfolder, at least...so it's clear they belong together and not to confuse them with the (tons of) other config files!

**Comment by mathias-luedtke** [comment]
> How about keeping the moveit_config files grouped in a subfolder, at least...

+1
I am not sure why it was moved into cob_hardware_config. I thought we move `cob_moveit_config` (without the launch files) into `cob_robots`..

**Comment by benmaidel** [comment]
> How about keeping the moveit_config files grouped in a subfolder, at least...

+1

**Comment by NadiaHG** [comment]
The suggested changes are done and also the cob_manipulation PRs adapted:
https://github.com/ipa320/cob_manipulation/pull/86  --> move the config files to cob_hardware_config
https://github.com/ipa320/cob_manipulation/pull/87/files --> move the config files to cob_hardware_config and rename the cob_moveit_config to cob_moveit_bringup

**Comment by fmessmer** [comment]
closing as it is handled in related PRs https://github.com/ipa320/cob_manipulation/pull/88 https://github.com/ipa320/cob_manipulation/pull/91 as well as https://github.com/ipa320/cob_robots/pull/573 https://github.com/ipa320/cob_robots/pull/567
